nearcore/src/config.rs

```diff
@@ -1327,39 +1336,93 @@ pub fn load_config(
     dir: &Path,
     genesis_validation: GenesisValidationMode,
 ) -> anyhow::Result<NearConfig> {
-    let config = Config::from_file(&dir.join(CONFIG_FILENAME))?;
-    let genesis_file = dir.join(&config.genesis_file);
+    let genesis_validation = match genesis_validation {
+        GenesisValidationMode::Full => GenesisValidationMode::Full,
+        GenesisValidationMode::UnsafeFast => GenesisValidationMode::UnsafeFast,
+    };
+    let mut validation_errors = ValidationErrors::new();
+
+    // if config.json has file issues, the program will directly panic
+    let config = Config::from_file_skip_validation(&dir.join(CONFIG_FILENAME))?;
+    // do config.json validation later so that genesis_file, validator_file and genesis_file can be validated before program panic
+    config.validate().map_or_else(|e| validation_errors.push_errors(e), |_| ());
+
     let validator_file = dir.join(&config.validator_key_file);
     let validator_signer = if validator_file.exists() {
-        let signer = InMemoryValidatorSigner::from_file(&validator_file).with_context(|| {
-            format!("Failed initializing validator signer from {}", validator_file.display())
-        })?;
-        Some(Arc::new(signer) as Arc<dyn ValidatorSigner>)
+        match InMemoryValidatorSigner::from_file(&validator_file) {
+            Ok(signer) => Some(Arc::new(signer) as Arc<dyn ValidatorSigner>),
+            Err(_) => None,
+        }
     } else {
+        let error_message =
+            format!("validator key file does not exist at the path {}", validator_file.display());
```

Comment:
yeah makes sense to not throw error here, also read the master branch code again, the logic there also does not throw error in case file does not exist, simply use a None for signer value.